### PR TITLE
Updates to PID

### PR DESCRIPTION
**Pull request by pkaig**

### CODE REVIEW

File: js/header_dialog.js [hunk 1]
```diff
@@ -964,6 +1030,22 @@ function HeaderDialog(dialog, onSave) {
...
+                // Scan the pid_bandwidth table
+                $(".pid_bandwidth input").each(function() {
+                        if($(this).val()!=null) {
+                                if($(this).attr('decPl')!=null) {
+                                        var matches=$(this).attr('name').match(/(.+)\[(\d+)\]/);
+                                        if(matches!=null) {
+                                                if(newSysConfig[matches[1]]==null) newSysConfig[matches[1]] = [null, null, null];
+                                                var newArray = newSysConfig[matches[1]];
+                                                newArray[matches[2]] = (parseFloat($(this).val()) * Math.pow(10, $(this).attr('decPl')));
+                                        } else (parseFloat($(this).val()) * Math.pow(10, $(this).attr('decPl')));
```
Comment (rotorflight) [inline js/header_dialog.js:1042]: what is this line doing? the () instead of {}?

File: js/header_dialog.js [hunk 2]
```diff
@@ -964,6 +1030,22 @@ function HeaderDialog(dialog, onSave) {
                         }
                 });
 
+                // Scan the pid_bandwidth table
+                $(".pid_bandwidth input").each(function() {
+                        if($(this).val()!=null) {
+                                if($(this).attr('decPl')!=null) {
+                                        var matches=$(this).attr('name').match(/(.+)\[(\d+)\]/);
```
Comment (rotorflight) [inline js/header_dialog.js:1037]: Be very careful using 'var'.
This should be 'let'

File: js/header_dialog.js [hunk 3]
```diff
@@ -176,8 +163,7 @@ function HeaderDialog(dialog, onSave) {
                         } else {
                                 parameterElem.addClass('missing');
                         }
-
-    }
+        }
 
     function setParameter(name, data, decimalPlaces) {
             var parameterElem = $('.parameter td[name="' + name + '"]');
```
Comment (rotorflight) [inline js/header_dialog.js:169]: const, not var

File: js/header_dialog.js [hunk 4]
```diff
@@ -880,9 +945,10 @@ function HeaderDialog(dialog, onSave) {
             setParameter('motor_poles'                            ,sysConfig.motor_poles, 0);
 
                 /* Booleans */
-        setCheckbox('gyro_cal_on_first_arm'                ,sysConfig.gyro_cal_on_first_arm);
-        setCheckbox('vbat_pid_compensation'                ,sysConfig.vbat_pid_compensation);
-        setCheckbox('rc_smoothing'                                ,sysConfig.rc_smoothing);
+        setCheckbox('gyro_cal_on_first_arm'     ,               sysConfig.gyro_cal_on_first_arm);
+        setCheckbox('vbat_pid_compensation'     ,               sysConfig.vbat_pid_compensation);
```
Comment (rotorflight) [inline js/header_dialog.js:949]: Is this still used?

File: js/header_dialog.js (hunk 2 above)
Comment (rotorflight) [inline js/header_dialog.js:1037]: const, not var

File: js/header_dialog.js (hunk 4 above)
Comment (pkaig) [reply js/header_dialog.js:949]: Neither vbat_pid_compensation or rc_smoothing are in the log file. There are a lot of parameters that need to be removed. I will review and compare the full list in a later PR.

File: js/header_dialog.js (hunk 3 above)
Comment (pkaig) [reply js/header_dialog.js:169]: Will update

File: js/header_dialog.js (hunk 1 above)
Comment (pkaig) [reply js/header_dialog.js:1042]: This function is not required and does not get used. Removing.

File: js/header_dialog.js (hunk 2 above)
Comment (pkaig) [reply js/header_dialog.js:1037]: This function is not required and does not get used. Removing.

File: js/header_dialog.js [hunk 5]
```diff
@@ -964,6 +1030,22 @@ function HeaderDialog(dialog, onSave) {
                         }
                 });
 
+                // Scan the pid_bandwidth table
+                $(".pid_bandwidth input").each(function() {
+                        if($(this).val()!=null) {
+                                if($(this).attr('decPl')!=null) {
```
[the remark this answers is not in the document]
Comment (pkaig) [reply js/header_dialog.js:1036]: This function is not required and does not get used. Removing.

File: js/header_dialog.js [hunk 6]
```diff
@@ -225,54 +209,133 @@ function HeaderDialog(dialog, onSave) {
...
+                                                break;
+                                        }
+                                } else $(this).closest('tr').addClass('missing');
+            })
+        }
+
+        function populateBandwidth(name, data) {
+                let i = 0;
+        const nameElem = $('.pid_bandwidth .' + name + ' input');
+        nameElem.each(function () {
```
Comment (rotorflight) [inline js/header_dialog.js:294]: The indentation looks weird here. I wonder what's going on... using tabs instaead of spaces maybe?

File: js/header_dialog.js [hunk 7]
```diff
@@ -176,12 +162,11 @@ function HeaderDialog(dialog, onSave) {
...
                         }
-
-    }
+        }
 
     function setParameter(name, data, decimalPlaces) {
-            var parameterElem = $('.parameter td[name="' + name + '"]');
-                var nameElem = $('input', parameterElem);
+            const parameterElem = $('.parameter td[name="' + name + '"]');
+                const nameElem = $('input', parameterElem);
```
Comment (rotorflight) [inline js/header_dialog.js:169]: Weird indentation

File: js/header_dialog.js [hunk 8]
```diff
@@ -225,54 +209,133 @@ function HeaderDialog(dialog, onSave) {
         }
 
         function populatePID(name, data) {
-                var i = 0;
-        var nameElem = $('.pid_tuning .' + name + ' input');
+                let i = 0;
+        const nameElem = $('.pid_tuning .' + name + ' input');
```
Comment (rotorflight) [inline js/header_dialog.js:213]: Intendeation

File: js/header_dialog.js [hunk 9]
```diff
@@ -225,54 +209,133 @@ function HeaderDialog(dialog, onSave) {
...
+                                                i++;
+                                                break;
+                                        }
+                                } else $(this).closest('tr').addClass('missing');
+            })
+        }
+
+        function populateBandwidth(name, data) {
+                let i = 0;
+        const nameElem = $('.pid_bandwidth .' + name + ' input');
```
Comment (rotorflight) [inline js/header_dialog.js:293]: indentation

**Review by rotorflight — CHANGES_REQUESTED** [review]
Check indentation

**Pull request closed without merging**